Source/WebCore/platform/mock/mediasource/MockMediaPlayerMediaSource.h

```diff
@@ -31,14 +31,18 @@
 #include "MediaPlayerPrivate.h"
 #include <wtf/Logger.h>
 #include <wtf/MediaTime.h>
+#include <wtf/RefCounted.h>
 #include <wtf/WeakPtr.h>
 
 namespace WebCore {
 
 class MediaSource;
 class MockMediaSourcePrivate;
 
-class MockMediaPlayerMediaSource final : public MediaPlayerPrivateInterface, public CanMakeWeakPtr<MockMediaPlayerMediaSource> {
+class MockMediaPlayerMediaSource final
+    : public MediaPlayerPrivateInterface
+    , public RefCounted<MockMediaPlayerMediaSource>
```

Comment:
Could you clarify why we can't have MediaPlayerPrivateInterface subclass RefCounted, instead of each of the subclasses? I went through the whole patch quickly and it wasn't directly obvious.

Reply:
Because I plan on making `MediaPlayerPrivateAVFoundation` inherit from `ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr` in a future patch, and I'd rather not create a diamond inheritance problem.

Reply:
(Maybe I should just do that for all subclasses that are already `CanMakeWeakPtr`.)